build_environment.bat

```diff
@@ -23,106 +159,154 @@ if %CONFIG_RELEASE% (
+SET INCLUDE_DIR=%ROOT_DIR%\%FOLDER_PLATFORM%\%BUILDTYPEDIR%\include\
 SET BUILD_DIR=%CD%\build\
 
-IF "%4" == "nuke" (
-  echo Deleting %LIB_DIR% and %INCLUDE_DIR%
-  rd /S /Q %LIB_DIR%
-  rd /S /Q %INCLUDE_DIR%
-  rem Protoc.exe comes with the repo, so can't delete BIN_DIR
-  rem   (Why don't we build Protoc? -Sean)
-  rem rd /S /Q %BIN_DIR%
-)
-
-set OLDPATH=%PATH%
+REM Everyting prepared. Setup the compiler.
 if %MACHINE_X86% (
-  rem If build OS is 64-bit:
-rem  call "%BUILDTOOLS_PATH%\vcbuildtools.bat" amd64_x86
+IF %OS_IS_64BIT% (
 call "%BUILDTOOLS_PATH%\%BUILDTOOLS_SCRIPT%" amd64_x86
-  rem If build OS is 32-bit, uncomment the following:
-  rem call "%BUILDTOOLS_PATH%\vcbuildtools.bat" x86
 ) else (
-  rem If build OS is 64-bit:
-rem  call "%BUILDTOOLS_PATH%\vcbuildtools.bat" amd64
+call "%BUILDTOOLS_PATH%\%BUILDTOOLS_SCRIPT%" x86
+)
+REM END OS_IS_64BIT
+) else ( REM NOT MACHINE_X86
+IF %OS_IS_64BIT% (
 call "%BUILDTOOLS_PATH%\%BUILDTOOLS_SCRIPT%" amd64
-  rem If build OS is 32-bit, uncomment the following:
-  rem call "%BUILDTOOLS_PATH%\vcbuildtools.bat" x86_amd64
+) else (
+call "%BUILDTOOLS_PATH%\%BUILDTOOLS_SCRIPT%" x86_amd64
 )
+REM END OS_IS_64
+)
+REM END MACHINE_X86
 
-md %LIB_DIR%
-md %INCLUDE_DIR%
-md %BIN_DIR%
-
+REM The Visual C++ compiler (cl.exe) recognizes certain environment variables, specifically LIB, LIBPATH, PATH, and INCLUDE
 rem Use our directories as well. Needed for a number of dependencies to find zlib, sqlite and xiph headers, including Qt.
 set INCLUDE=%INCLUDE%;%INCLUDE_DIR%
 set LIB=%LIB%;%LIB_DIR%
 set UseEnv=true
 rem Qt may need this
 set LIBPATH=%LIBPATH%;%LIB_DIR%
 
-call build_sqlite3.bat
-call build_zlib.bat 
-call build_pthreads.bat
-call build_protobuf.bat
-call build_portmidi.bat
-call build_libid3tag.bat REM depends on zlib
-call build_libmad.bat
-call build_libogg.bat
-call build_libopus.bat REM depends on libogg
-call build_libvorbis.bat
-call build_openssl.bat
-call build_libshout.bat REM may soon depend on openssl
-call build_libflac.bat
-call build_libsndfile.bat
-call build_fftw3.bat
-call build_rubberband.bat REM depends on fftw3.h
-call build_portaudio.bat
-call build_hss1394.bat
-call build_chromaprint.bat REM depends on fftw3
-call build_taglib.bat REM depends on zlib 
-REM We do not distribute LAME with Mixxx. If you wish to build it locally, uncomment.
-REM call build_lame.bat
-call build_qt4.bat REM depends on sqlite3, zlib and openssl
+
+REM ==================================
+REM Deletion and directory preparation
+REM ==================================
+
+IF %DELETE_OLD% (
+  echo Deleting %LIB_DIR% and %INCLUDE_DIR%
+  rd /S /Q %LIB_DIR%
+  rd /S /Q %INCLUDE_DIR%
+  rd /S /Q %BIN_DIR%
+)
+md %LIB_DIR%
+md %INCLUDE_DIR%
+md %BIN_DIR%
+copy %CD%\bin\*.* %BIN_DIR%
+
+REM ==================================
+REM Script execution to compile each lib
+REM ==================================
+
+SETLOCAL EnableDelayedExpansion
+SET PROCESSALL=1==1
+IF [%JUMPIDX%] == [] (
+  SET JUMPIDX = 1
```

Comment:
I had to change this to ```SET JUMPIDX=1```, otherwise the comparison in the for loop was a syntax error.